src/plasma/plasma_manager.cc

```diff
@@ -378,23 +382,25 @@ void remove_wait_request(PlasmaManagerState *manager_state,
     CHECK(event_loop_remove_timer(manager_state->loop, wait_req->timer) ==
           AE_OK);
   }
-  free(wait_req->object_requests);
+  delete wait_req->object_requests;
   free(wait_req);
 }
 
 void return_from_wait(PlasmaManagerState *manager_state,
                       WaitRequest *wait_req) {
   /* Send the reply to the client. */
   warn_if_sigpipe(plasma_send_WaitReply(
-                      wait_req->client_conn->fd, manager_state->builder,
-                      wait_req->object_requests, wait_req->num_object_requests),
+                  wait_req->client_conn->fd, manager_state->builder,
+                  *(wait_req->object_requests), wait_req->num_object_requests),
                   wait_req->client_conn->fd);
-  /* Remove the wait request from each of the relevant object_wait_requests hash
+  /* Iterate over all object IDs requested as part of this wait request.
+   * Remove the wait request from each of the relevant object_wait_requests hash
    * tables if it is present there. */
-  for (int i = 0; i < wait_req->num_object_requests; ++i) {
+  for (const auto &objreq_pair : *(wait_req->object_requests)) {
```

Comment:
Let's expand the `objreq` in the variable names to `object_request` or something like that